libs/genai/tests/integration_tests/test_chat_models.py

```diff
@@ -174,6 +175,39 @@ def test_chat_google_genai_invoke_with_modalities() -> None:
     _check_usage_metadata(result)
 
 
+def test_chat_google_genai_invoke_thinking() -> None:
+    """Test invoke thinking model from ChatGoogleGenerativeAI with
+    default thinking config"""
+    llm = ChatGoogleGenerativeAI(model=_THINKING_MODEL, thinking_budget=100)
+
+    result = llm.invoke(
+        "How many O's are in Google? Please tell me how you double checked the result",
+    )
+
+    assert isinstance(result, AIMessage)
+    assert isinstance(result.content, str)
+
+    _check_usage_metadata(result)
+
+    assert result.usage_metadata["output_token_details"]["reasoning"] > 0
+    
+
+def test_chat_google_genai_invoke_thinking_disabled() -> None:
+    """Test invoke thinking model from ChatGoogleGenerativeAI with
+    default thinking config"""
+    llm = ChatGoogleGenerativeAI(model=_THINKING_MODEL, thinking_budget=0)
+
+    result = llm.invoke(
+        "How many O's are in Google? Please tell me how you double checked the result",
+    )
+
+    assert isinstance(result, AIMessage)
+    assert isinstance(result.content, str)
+
+    _check_usage_metadata(result)
+
+    assert "output_token_details" not in result.usage_metadata
```

Comment:
we don't need a separate test for that, add it to an existing integration test, please

Reply:
thinking model by default performs auto reasoning, so this test explicitly disables the thinking by setting the budget to 0.